src/runtime/forester/flow.rs

```diff
@@ -299,6 +365,10 @@ pub fn monitor(
 pub enum FlowDecision {
     PopNode(RNodeState),
     Stay(RNodeState),
+    Halt {
+        new_state: RNodeState,
```

Comment:
since it is an internal class that i used here I would suggest to 
```rust
type HaltingChildCursor = i64;
Halt(RNodeState,HaltingChildCursor), 
```
or simply
```
Halt(RNodeState,i64), 
```

Reply:
Fair enough, that's reasonable.

I really like the self-documenting nature of explicitly labelling the second element as the cursor for the halting child, so I'm going to go for your first suggest. I'm worried just `i64` will be harder to understand and maintain going forwards.

The code has been updated.